solutions/USACO jan19-mountain-view.mdx

```diff
@@ -0,0 +1,75 @@
+---
+id: usaco-896
+source: USACO Silver 2019 January
+title: Mountain View
+author: Jesse Choe
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_prob3_silver_dec20.html)
+
+**Time Complexity:** $\mathcal{O}(n\log n)$
+
+<LanguageSection>
+<CPPSection>
+
+## Sorting Struct Using Comparator Function
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define FastIO ios_base::sync_with_stdio(0); cin.tie(0)
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sor(x) sort(all(x))
+#define ins insert
+#define f first
+#define s second
+#define sz(x) (int)(x).size()
+#define len(x) (int)(x).length()
+#define FOR(i, a, b) for(int i=a; i<b; ++i)
+#define ROF(i, b, a) for(int i=(b)-1; i>=a; --i)
+
+void setIO(string fileName = ""){
+	FastIO;
+	if(len(fileName)){
+		freopen((fileName+".in").c_str(), "r", stdin);
+		freopen((fileName+".out").c_str(), "w", stdout);
+	}
+}
+
+struct Mountain {
+    int l, r;
+};
```

Comment:
use pair 👀

Reply:
lol the point of this sol was to use a struct

Reply:
this is kinda a bad example of struct with custom comparator then. its just default sort which is eh...

Reply:
^^ imo if the struct doesn't add additional functionality/readability to STL then i think it's better to j/ leave it.